version/version.go

```diff
@@ -31,6 +33,17 @@ var (
 
 	// GoVersion is Go tree's version.
 	GoVersion = runtime.Version()
+
+	// UserAgent is the default User-Agent header for http-requests such as
+	// authenticating and pulling images from registries using the
+	// core/remotes/docker package. This variable is filled in at linking
+	// time, and users of containerd as a go module must set this variable
+	// to a value matching their product, following the guidelines outlined
+	// in [RFC9110, section 10.1.5], or to configure the User-Agent header
+	// through other means.
+	//
+	// [RFC9110, section 10.1.5]: https://httpwg.org/specs/rfc9110.html#field.user-agent
+	UserAgent = "containerd/2.0.0-rc.3+unknown"
```

Comment:
FWIW I tried making this `UserAgent = "containerd/" + Version` but unfortunately that prevents it being overridden through `-X`

Reply:
I'm also curious if we should make this use `containerd-client/` (or something along those lines) so that the default doesn't try to pretend it's "containerd".

Reply:
Maybe more complicated but make `UserAgent` a func and define a `UserAgentOverride` and maybe `UserAgentPrefix`? If override is set it returns that, otherwise it concats the prefix with the version. Then we could possibly use separate user agent for ctr and containerd daemon easily.